services/app/src/contexts/ChessBoardContext.jsx

```diff
@@ -0,0 +1,34 @@
+/* eslint-disable react/jsx-no-constructed-context-values */
+/* eslint-disable react/prop-types */
+import React, { createContext, useState } from 'react';
+import { useChessBoard } from '../hooks/useChessBoard';
+
+export const ChessBoardContext = createContext();
+
+const ChessBoardContextProvider = ({ children }) => {
+  const [selectedItem, setSelectedItem] = useState('');
+  const [board, setBoard] = useChessBoard();
+
+  const placeItemOnBoard = (row, tile) => {
+    if (selectedItem) {
+      const copyOfBoard = [...board];
+      copyOfBoard[row][tile].items.push(selectedItem);
+      setBoard(copyOfBoard);
+      setSelectedItem('');
+    }
+  };
+
+  return (
+    <ChessBoardContext.Provider
+      value={{
+        setSelectedItem,
+        selectedItem,
+        placeItemOnBoard,
+        board,
+      }}
+    >
+      {children}
+    </ChessBoardContext.Provider>
+  );
+};
```

Comment:
Let's add a new named export for using this context:
```
const useChessBoardContext = () => {
  return useContext(ChessBoardContext);
};
```
and use this custom hook when we need `ChessBoardContext`